src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -535,6 +545,40 @@ protected void parseCreateField(ParseContext context, int dimension) throws IOEx
         context.path().remove();
     }
 
+    // Verify mapping and return true if it is a "faiss" Index using "sq" encoder
+    // of type "fp16"
+    protected boolean isFaissSQfp16(KNNMethodContext knnMethodContext) {
+        if (knnMethodContext != null) {
+            if (knnMethodContext.getKnnEngine().getName().equals(FAISS_NAME)
```

Comment:
Too many nested. Can you flip if to reduce nested ifs

Reply:
updated